examples/chart/teleport-agent/README.md

```diff
@@ -0,0 +1,35 @@
+# Teleport Kubernetes Agent
+
+This chart is a minimal Teleport agent used to register a Kubernetes cluster
+with an existing Teleport cluster.
+
+To use it, you will need:
+- an existing Teleport cluster (at least proxy and auth services)
+- a reachable proxy endpoint (`$PROXY_ENDPOINT`)
+- a [static join
+  token](https://goteleport.com/teleport/docs/admin-guide/#adding-nodes-to-the-cluster)
+  for this Teleport cluster (`$JOIN_TOKEN`)
+  - this chart does not currently support dynamic join tokens; please [file an
```

Comment:
What's the reason for it not supporting dynamic join tokens?

Reply:
I think it should work as long as the user pre-generates the token with `tctl` and passes it in here...

Reply:
Ah, it doesn't work because there's no persistent storage so Teleport will have to rejoin and regenerate certs every time the pod restarts.

A user could work around this by exporting a dynamic join token as a k8s secret and having the chart mount this secret as a file inside the pod which Teleport reads on startup. Whenever the secret is updated, the file will change and the new secret will be used on restart. We could add instructions to the README on how to do this.

Theoretically, though, a static join token with sufficient entropy which is only valid for the `Kubernetes` role should be secure enough.

Reply:
Yeah, it's all about long-term issues due to pod restarts/reschedules.
Static token is always worse (even with large entropy), but we currently don't have a good mechanism to distribute dynamic tokens within k8s. I want to keep the UX as simple as possible (fire-and-forget 1-liner).

We should think about some tighter integration between teleport and k8s, for example using the k8s API as a backend for some essential teleport state (e.g. post-bootstrap credentials, process UUID, etc).